src/main/java/io/aiven/kafka/connect/http/sender/ApiKeyAccessTokenHttpRequestBuilder.java

```diff
@@ -0,0 +1,78 @@
+package io.aiven.kafka.connect.http.sender;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.aiven.kafka.connect.http.config.HttpSinkConfig;
+
+import java.net.http.HttpRequest;
+import java.time.Duration;
+import java.util.Objects;
+
+public class ApiKeyAccessTokenHttpRequestBuilder implements HttpRequestBuilder {
+
+    ApiKeyAccessTokenHttpRequestBuilder() {
+    }
+
+    @Override
+    public HttpRequest.Builder build(HttpSinkConfig config) {
+        Objects.requireNonNull(config, "config");
+        Objects.requireNonNull(config.oauth2AccessTokenUri(), "oauth2AccessTokenUri");
+        Objects.requireNonNull(config.oauth2ClientId(), "oauth2ClientId");
+        Objects.requireNonNull(config.oauth2ClientSecret(), "oauth2ClientSecret");
+
+        final var accessTokenRequestBuilder = HttpRequest
+                .newBuilder(config.oauth2AccessTokenUri())
+                .timeout(Duration.ofSeconds(config.httpTimeout()))
+                .header(HEADER_CONTENT_TYPE, "application/json");
+
+        AccessTokenRequest accessTokenRequest = new AccessTokenRequest();
+        accessTokenRequest.setType("api-key");
+        accessTokenRequest.setKey(config.oauth2ClientId());
+        accessTokenRequest.setSecret(config.oauth2ClientSecret().value());
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        try {
+            String requestBody = objectMapper
+                    .writerWithDefaultPrettyPrinter()
+                    .writeValueAsString(accessTokenRequest);
+            return accessTokenRequestBuilder
+                    .POST(HttpRequest.BodyPublishers.ofString(requestBody));
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    static class AccessTokenRequest {
```

Comment:
this inner class should be private and immutable by setting all of this field to final and get rid of the setters

```
final String type;
        final String key;
        final String secret;

        public AccessTokenRequest(String type, String key, String secret) {
            this.type = type;
            this.key = key;
            this.secret = secret;
        }
```